src/components/structuredefSettings/StructuredefSettings.tsx

```diff
@@ -53,33 +47,24 @@ const StructuredefSettings: React.FC<StructuredefSettingsProps> = ({
       e.preventDefault();
       const structureDefinitonToEdit = { ...structureDefinition };
       // content required
-      structureDefinitonToEdit.url = url as Uri.AsObject;
-      structureDefinitonToEdit.name = name as String.AsObject;
-      structureDefinitonToEdit.kind = kind as StructureDefinition.KindCode.AsObject;
-      structureDefinitonToEdit.abstract = abstract as Boolean.AsObject;
-      structureDefinitonToEdit.status = status as StructureDefinition.StatusCode.AsObject;
+      structureDefinitonToEdit.url = url as string;
```

Comment:
why do you need to "cast" the type ? shouldn't typescript be aware that `url` is a string ?